src/main/drivers/gyro_sync.c

```diff
@@ -30,29 +30,21 @@
 
 extern gyro_t gyro;
 
-uint32_t targetLooptime;
 static uint8_t mpuDividerDrops;
 
-bool gyroSyncCheckUpdate(void)
+uint32_t gyroSetSampleRate(uint8_t lpf, uint8_t gyroSyncDenominator)
 {
-    return gyro.isDataReady && gyro.isDataReady();
-}
-
-void gyroSetSampleRate(uint32_t looptime, uint8_t lpf, uint8_t gyroSync, uint8_t gyroSyncDenominator)
-{
-    if (gyroSync) {
-        int gyroSamplePeriod;
-        if (lpf == 0) {
-            gyroSamplePeriod = 125;
-        } else {
-            gyroSamplePeriod = 1000;
-        }
-        mpuDividerDrops = gyroSyncDenominator - 1;
-        targetLooptime = gyroSyncDenominator * gyroSamplePeriod;
+    int gyroSamplePeriod;
+    if (lpf == GYRO_LPF_256HZ || lpf == GYRO_LPF_NONE) {
+        gyroSamplePeriod = 125;
     } else {
-        mpuDividerDrops = 0;
-        targetLooptime = looptime;
+        gyroSamplePeriod = 1000;
+        gyroSyncDenominator = 1; // Always full Sampling 1khz
     }
+
+    mpuDividerDrops = gyroSyncDenominator - 1;
+    const uint32_t targetLooptime = gyroSyncDenominator * gyroSamplePeriod;
```

Comment:
Most F3 targets can go lower. Technically F1 too with some optimalisations and acc disabled. Its a choice for user